application/src/Api/Representation/AbstractResourceEntityRepresentation.php

```diff
@@ -451,17 +451,42 @@ public function objectValues()
      * Options:
      *
      * - viewName: Name of view script, or a view model. Default "common/resource-values"
+     * - siteId: A site ID
      *
      * @param array $options
      * @return string
      */
     public function displayValues(array $options = [])
     {
         $options['viewName'] = $options['viewName'] ?? 'common/resource-values';
-        $partial = $this->getViewHelper('partial');
+        $options['siteId'] = $options['siteId'] ?? null;
+
+        $services = $this->getServiceLocator();
+        $values = $this->values();
+
+        if ($options['siteId'] && $services->get('Omeka\Settings\Site')->get('exclude_resources_not_in_site')) {
```

Comment:
This bit is a little hinky: it checks for a passed in site id but then checks against the current site's settings. Better probably to move the check out into the partial so you only pass the site id if the current site is configured to exclude.

This change would allow for what the siteId option would seem to imply, that you can pass a different site's id and have it work right, or that you could pass one from the admin side.